lxd/storage/backend_lxd.go

```diff
@@ -1386,12 +1386,16 @@ func (b *lxdBackend) RefreshCustomVolume(projectName string, srcProjectName stri
 
 // RefreshInstance synchronises one instance's volume (and optionally snapshots) over another.
 // Snapshots that are not present in the source but are in the destination are removed from the
-// destination if snapshots are included in the synchronisation.
+// destination if snapshots are included in the synchronisation. An empty srcSnapshots argument
+// indicates a volume-only refresh.
 func (b *lxdBackend) RefreshInstance(inst instance.Instance, src instance.Instance, srcSnapshots []instance.Instance, allowInconsistent bool, op *operations.Operation) error {
 	l := logger.AddContext(b.logger, logger.Ctx{"project": inst.Project().Name, "instance": inst.Name(), "src": src.Name(), "srcSnapshots": len(srcSnapshots)})
 	l.Debug("RefreshInstance started")
 	defer l.Debug("RefreshInstance finished")
 
+	// This indicates whether or not it's a volume-only refresh.
+	snapshots := len(srcSnapshots) > 0
```

Comment:
Do we need to differentiate between srcSnapshots being nil or empty?
Is there a difference for optimized refresh between there being no snapshots (but not in instance only mode) vs instance only mode?

Reply:
We don't need to differentiate, I checked that.

Reply:
Thanks.